src/k8s/pkg/k8sd/api/component.go

```diff
@@ -43,7 +43,7 @@ func putDNSComponent(s *state.State, r *http.Request) response.Response {
 
 	switch req.Status {
 	case api.ComponentEnable:
-		dnsIP, clusterDomain, err := component.EnableDNSComponent(snap, req.Config.ClusterDomain, req.Config.ServiceIP, req.Config.UpstreamNameservers)
+		dnsIP, clusterDomain, err := component.EnableDNSComponent(s.Context, snap, req.Config.ClusterDomain, req.Config.ServiceIP, req.Config.UpstreamNameservers)
```

Comment:
We should use the request context here and below: `r.Context`
This request context is cancelled if the user terminates the request. The state context is only cancelled if `k8sd` terminates.